Build fails on MacOS (Apple M1) with gcc-11 nix shell

On my Mac (Apple M1), a plain

```b
nix develop .#ci_gcc11
make mlkem
```

results in

```
...
  AS      test/build/mlkem512/mlkem/native/x86_64/src/ntt.S.o
  CC      test/build/mlkem512/mlkem/native/x86_64/src/rej_uniform_avx2.c.o
  AS      test/build/mlkem512/mlkem/native/x86_64/src/shuffle.S.o
  AR      test/build/libmlkem512.a
ld: unknown option: --whole-archive
collect2: error: ld returned 1 exit status
make: *** [mk/rules.mk:23: test/build/libmlkem512.a] Error 1
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -252,6 +252,8 @@ jobs:
            name: 'aarch64'
          - runner: ubuntu-latest
            name: 'x86_64'
+         - runner: macos-latest
+           name: 'macos'
         exclude:
           - {external: true,
              target: {
@@ -262,6 +264,7 @@ jobs:
     steps:
       - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
       - name: native build+functest (gcc-4.8)
+        if: ${{ matrix.target.runner != 'macos-latest' }}
         uses: ./.github/actions/multi-functest
         with:
           gh_token: ${{ secrets.GITHUB_TOKEN }}
@@ -272,6 +275,7 @@ jobs:
           acvp: false
           nix-shell: "ci_gcc48"
       - name: native build+functest (gcc-4.9)
+        if: ${{ matrix.target.runner != 'macos-latest' }}
         uses: ./.github/actions/multi-functest
         with:
           gh_token: ${{ secrets.GITHUB_TOKEN }}
@@ -282,6 +286,7 @@ jobs:
           acvp: false
           nix-shell: "ci_gcc49"
       - name: native build+functest (gcc-7)
+        if: ${{ matrix.target.runner != 'macos-latest' }}
         uses: ./.github/actions/multi-functest
         with:
           gh_token: ${{ secrets.GITHUB_TOKEN }}

--- a/mk/rules.mk
+++ b/mk/rules.mk
@@ -24,13 +24,13 @@ $(BUILD_DIR)/%.a: $(CONFIG)
 # 	$AR doesn't care about duplicated symbols, one can only find it out via actually linking.
 # 	The easiest one to do this that one can think of is to create a dummy C file with empty main function on the fly, pipe it to $CC and link with the built library
 	$(eval _LIB := $(subst $(BUILD_DIR)/lib,,$(@:%.a=%)))
-ifneq ($(findstring clang,$(shell $(CC) --version)),) # if CC is clang
+ifneq ($(findstring Darwin,$(HOST_PLATFORM)),) # if is on macOS
 	$(Q)echo "int main() {return 0;}" \
 		| $(CC) -x c - -L$(BUILD_DIR) \
 		 -all_load -Wl,-undefined,dynamic_lookup -l$(_LIB) \
 		 -Imlkem $(wildcard test/notrandombytes/*.c)
 	$(Q)rm -f a.out
-else                                                  # if CC is not clang
+else                                           # if not on macOS
 	$(Q)echo "int main() {return 0;}" \
 		| $(CC) -x c - -L$(BUILD_DIR) \
 		-Wl,--whole-archive,--unresolved-symbols=ignore-in-object-files -l$(_LIB) \

